internal/graph/check.go

```diff
@@ -1258,18 +1259,37 @@ func processUsersetMessage(message usersetMessage,
 
 func matchUsersetFromUserAndUsersetFromObject(ctx context.Context,
 	req *ResolveCheckRequest,
-	userToUsersetMessageChan,
-	objectToUsersetMessageChan chan usersetMessage) (*ResolveCheckResponse, storage.SortedSet, storage.SortedSet, error) {
+	objectToUsersetMessageChan chan usersetMessage,
+	typesys *typesystem.TypeSystem,
+	ds storage.RelationshipTupleReader) (*ResolveCheckResponse, storage.SortedSet, storage.SortedSet, error) {
 	usersetFromUser := storage.NewSortedSet()
 	usersetFromObject := storage.NewSortedSet()
 
 	userToUsersetDone := false
 	objectToUsersetDone := false
 
+	// check to see if there are any nested userset assigned.  If not,
+	// we don't even need to check the user side.
+	objectToUsersetMessageInit, okInit := <-objectToUsersetMessageChan
+	if !okInit {
```

Comment:
This is important - we should not call streamedLookupUsersetForUser if we know there are no nested userset from the object side.